types/container/config.go

```diff
@@ -3,8 +3,29 @@ package container
 import (
 	"github.com/docker/engine-api/types/strslice"
 	"github.com/docker/go-connections/nat"
+	"time"
 )
 
+// HealthConfig holds configuration settings for the HEALTHCHECK feature.
+type HealthConfig struct {
+	// Test to perform to check container is healthy.
+	// An empty slice means to inherit the default.
+	// The options are:
+	// {} : inherit healthcheck
+	// {"NONE"} : disable healthcheck
+	// {"CMD", args...} : exec arguments directly
+	// {"CMD-SHELL", command} : run command with system's default shell
+	Test []string `json:",omitempty"`
+
+	// Zero means to inherit. Durations are expressed as integer nanoseconds.
+	Interval time.Duration `json:",omitempty"` // Time to wait between checks.
+	Timeout  time.Duration `json:",omitempty"` // Time to wait before considering the check to have hung.
+
+	// Number of consecutive failures needed to consider a container as unhealthy.
+	// Zero means inherit.
+	Retries int `json:",omitempty"`
```

Comment:
> I don't think this applies to fields within a struct, since they're not at the top level. Most other fields in the source don't repeat the field name in the comment.

It absolutely does. These are all exported names, even if they are struct fields.